### ISSUE TITLE
Accessibility and keyboard navigation?

### ISSUE THREAD

**Issue by inorton-entrust** [TomSchimansky/CustomTkinter#2786]
Hi, I've been reading #2712 and have some more general questions about accessibility.  I would like to use a screen reader on an CTk app, partly to automate testing and partly of course because of accessibility.

I am finding that I am unable to use tab / shift-tab to cycle between elements at all, or be able to activate buttons or enter text entry boxes.  Is there something I can do to fix this? I am willing to contribute code.

Thanks

**Comment by FedericoSpada** [comment]
Hi,
Tab key management is completely missing and should be developed almost from scratch.
If you want to try to implement it, feel free to do so and open a Pull Request (start from the develop branch that is ahead of the main one).

I would say that the developer should be able to assign a Tab order to the various widgets: maybe via an attribute to each widget, or maybe by providing a proper list to a special method of the parent widget. A default behavior should also be implemented, probably based on the creation order. It should be possible to skip a widget altogether. Tab key on a CTkTextbox should actually write a Tab char.

**Comment by inorton-entrust** [comment]
@FedericoSpada after logging the issue I did actually manage to make tabbing around my app work. I wasn't able to explicitly set the order (I'm stuck with the order they appear via pack()) but I do have an app I can entirely navigate with keyboard only. I'll see if I can put together a PR.  But my app only used CtkButton, ttk.ComboBox, ttk.Treeview and labels etc
